src/index.ts

```diff
@@ -140,6 +143,31 @@ async function main() {
       );
     });
 
+    const prompts = PromptFactory.getPrompts();
+    // Register prompts (add this block)
+    prompts.forEach((prompt) => {
+      server.prompt(prompt.name, (args) => {
+        // Remove extra properties (like 'signal') from args before passing to fillTemplate
+        // Convert all values to string for template replacement
+        const inputArgs = Object.fromEntries(
+          Object.entries(args)
+            .filter(([key]) => key !== "signal")
+            .map(([k, v]) => [k, v == null ? "" : String(v)]), // Ensure all values are strings
+        );
+        return {
+          messages: [
+            {
+              role: "user",
+              content: {
+                type: "text",
+                text: fillTemplate(prompt.description, inputArgs),
+              },
+            },
+          ],
+        };
+      });
+    });
```

Comment:
Something here feels weird, I'm not entirely sure why `fillTemplate` is necessary since we could instead, like we do with tools, include the argsSchema when creating the prompt.

An example of a prompt that has a schema:
```ts
server.prompt(
  "review-code",
  { code: z.string() },
  ({ code }) => ({
    messages: [{
      role: "user",
      content: {
        type: "text",
        text: `Please review this code:\n\n${code}`
      }
    }]
  })
);
```

So we shouldn't need to perform the swap ourselves and remove the `fillTemplate` code.